sjsonnet/src/sjsonnet/Evaluator.scala

```diff
@@ -762,8 +768,7 @@ class Evaluator(
     for (s <- visitComp(e.first :: e.rest, Array(compScope))) {
       visitExpr(e.key)(s) match {
         case Val.Str(_, k) =>
-          val prev_length = builder.size()
-          builder.put(
+          val previousValue = builder.put(
```

Comment:
This change technically isn't a bugfix, but is a small optimization to bring the existing comprehension evaluation code into line with the approach used for object bodies.